### PR TITLE
Add insert, remove and update methods to multi_index (and index)

### LINKED ISSUE #135: Support standard naming conventions for databases
Right now we are using `emplace`, `modify`, and `erase` for the manipulation of rows in `multi_index` tables. 

Since we are expecting `multi_index` to be used in a similar way to a database within smart contracts, it would be easier for developers to get started with Antelope if our nomenclature mimicked that of other databases they are familiar with (like MySQL, Postgre, MongoDB, SQLite, etc) and had alias support for:
- Insert
- Update
- Delete

I think that the understandability and memorability of our language would improve for anyone who doesn't come from C++, Rust, or D. 

```cpp
users.insert(ram_payer, [&](auto& row){ ... });
users.update(user, ram_payer, [&](auto& row){ ... });
users.delete(user);
```

### PR DESCRIPTION
**Pull request by mikelik**
## Change Description

Resolves #135 
Add insert, remove and update methods to multi_index

## API Changes
- [x] API Changes

For multi_index there are 3 new methods: insert (alias for emplace), remove (alias for erase), update (alias for modify).

## Documentation Additions
- [ ] Documentation Additions

If there is documentation for multi_index 3 new methods should be added. It might be in separate repo `docs`.

### CODE REVIEW

File: libraries/eosiolib/contracts/eosio/multi_index.hpp [hunk 1]
```diff
@@ -732,11 +732,21 @@ class multi_index
...
+               modify( itr, payer, std::forward<Lambda>(updater) );
+            }
+
             template<typename Lambda>
             void modify( const T& obj, eosio::name payer, Lambda&& updater ) {
                _multidx->modify( obj, payer, std::forward<Lambda&&>(updater) );
             }
 
+            template<typename Lambda>
+            void update( const T& obj, eosio::name payer, Lambda&& updater ) {
```
Comment (dimas1185) [inline libraries/eosiolib/contracts/eosio/multi_index.hpp:746]: `modify` has a small design flaw where it accepts `const` object and later modifies it using `const_cast`.
to not break existing contracts code we may leave `modify` as it is but for new interface I suggest we remove const-ness, it just confuses.

File: libraries/eosiolib/contracts/eosio/multi_index.hpp [hunk 2]
```diff
@@ -841,6 +855,56 @@ class multi_index
...
+         using namespace _multi_index_detail;
+
+         auto& objitem = static_cast<item&>(obj);         
+         eosio::check( objitem.__idx == this, "object passed to modify is not in multi_index" );
+
+         eosio::check( _code == current_receiver(), "cannot modify objects in table of another contract" ); // Quick fix for mutating db using multi_index that shouldn't allow mutation. Real fix can come in RC2.
+
+         auto secondary_keys = make_extractor_tuple::get_extractor_tuple(indices_type{}, obj);
+
+         uint64_t pk = _multi_index_detail::to_raw_key(obj.primary_key());
```
Comment (jolly-fellow) [inline libraries/eosiolib/contracts/eosio/multi_index.hpp:869]: IMHO if you declare "using namespace _multi_index_detail;" it doesn't make sense to declare it here again.

File: libraries/eosiolib/contracts/eosio/multi_index.hpp [hunk 3]
```diff
@@ -841,6 +855,56 @@ class multi_index
...
+
+         updater( obj );
+
+         eosio::check( pk == _multi_index_detail::to_raw_key(obj.primary_key()), "updater cannot change primary key when modifying an object" );
```
Comment (jolly-fellow) [inline libraries/eosiolib/contracts/eosio/multi_index.hpp:873]: IMHO it doesn't make sense to declare  _multi_index_detail here again.

**Review by dimas1185 — APPROVED** [review]
please address @jolly-fellow comments, not critical but will reduce verbosity

File: libraries/eosiolib/contracts/eosio/multi_index.hpp (hunk 2 above)
Comment (mikelik) [reply libraries/eosiolib/contracts/eosio/multi_index.hpp]: fixed

**Review by jolly-fellow — APPROVED** [review]

**Comment by ericpassmore** [comment]
Note:start
group: STABILITY
category: INTERNALS
summary: Add alternate names for operations to insert, remove, and update indexes.
Note:end

**Pull request merged**